src/main/java/seedu/address/logic/commands/lesson/MarkLessonParticipationCommand.java

```diff
@@ -0,0 +1,132 @@
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.CommandType;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.lesson.Lesson;
+import seedu.address.model.student.Name;
+import seedu.address.model.student.Student;
+import seedu.address.model.student.exceptions.StudentNotFoundException;
+
+/**
+ * Marks chosen students' participation for a particular Lesson.
+ */
+public class MarkLessonParticipationCommand extends Command {
+
+    public static final String COMMAND_WORD = "markp";
+    public static final CommandType COMMAND_TYPE = CommandType.LESSON;
+    public static final int LOWER_BOUND = 0;
+    public static final int UPPER_BOUND = 100;
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sets the participation of student(s) "
+            + "in a lesson at the chosen index in the lesson list to the specified value. "
+            + "\nIf their participation is set to a positive integer, also sets their attendance to true."
+            + "\nParameters: LESSON_INDEX (must be a positive integer) "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_POINTS + "PARTICIPATION (integer from 0-100)"
+            + "\nExample: " + COMMAND_WORD + " 1 n/John Doe pt/1";
+
+    public static final String MESSAGE_SUCCESS = "Marked the participation of %s as %s";
+    public static final String MESSAGE_STUDENT_NOT_FOUND_IN_ADDRESS_BOOK = "Student not found in TAHub: %s";
+    public static final String MESSAGE_STUDENT_NOT_FOUND_IN_LESSON = "Student not found in the lesson: %s";
+    public static final String MESSAGE_INVALID_PARTICIPATION =
+            "Participation should be between 0-100 inclusive.";
+
+    private final Index index;
+    private final List<Name> studentNames;
+    private final int participationScore;
+    private final Logger logger = LogsCenter.getLogger(MarkLessonParticipationCommand.class);
+
+    /**
+     * @param index of the lesson in the filtered lesson list
+     * @param studentNames list of student names to target
+     * @param participationScore the student(s)' participation score
+     */
+    public MarkLessonParticipationCommand(Index index, List<Name> studentNames, int participationScore) {
+        requireAllNonNull(index, studentNames);
+        this.index = index;
+        this.studentNames = studentNames;
+        this.participationScore = participationScore;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        logger.info("Setting lesson participation of students");
+        List<Lesson> lastShownLessonList = model.getFilteredLessonList();
+
+        if (index.getZeroBased() >= lastShownLessonList.size()) {
+            throw new CommandException(String.format(MESSAGE_INVALID_LESSON_DISPLAYED_INDEX, index.getOneBased()));
+        }
+
+        if (!isWithinBounds(participationScore)) {
```

Comment:
Can consider doing validity checking for participation score in ParserUtil because checking validity of Name, Phone etc are checked there.

Reply:
I intended for `parsePoints` in parserUtil to be a generic method to parse "points" so it could be reused for other things like assignment points (marks). However, seeing as we probably aren't going to work on such an extension, your suggestion could be a cleaner solution.